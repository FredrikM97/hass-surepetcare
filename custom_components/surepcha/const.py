""" "Constants for the Sure Petcare integration."""

DOMAIN = "surepcha"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
ENTRY_ID = "entry_id"
SCAN_INTERVAL = 300
POLLING_SPEED = "polling_speed"
LOCATION_INSIDE = "location_inside"
LOCATION_OUTSIDE = "location_outside"
OPTION_DEVICES = "devices"
DEVICE_OPTION = "device_option"
OPTIONS_FINISHED = "finished"
PRODUCT_ID = "product_id"
DEVICES = "devices"
NAME = "name"
