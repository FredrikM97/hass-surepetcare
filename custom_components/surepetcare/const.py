"""TODO."""

DOMAIN = "surepetcare"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
