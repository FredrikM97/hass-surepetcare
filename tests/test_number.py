import pytest
from unittest.mock import patch
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import initialize_entry

from pytest_homeassistant_custom_component.common import (
    MockConfigEntry,
    snapshot_platform,
)


@patch("custom_components.surepcha.PLATFORMS", [Platform.NUMBER])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


@patch("custom_components.surepcha.PLATFORMS", [Platform.NUMBER])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery_missing_entities(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry_missing_entities: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry_missing_entities, mock_devices, mock_pets
    )
    await snapshot_platform(
        hass, entity_registry, snapshot, mock_config_entry_missing_entities.entry_id
    )


@patch("custom_components.surepcha.PLATFORMS", [Platform.NUMBER])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_number_set_and_snapshot(
    hass,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
):
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await hass.async_block_till_done()

    for entity_id in hass.states.async_entity_ids("number"):
        state = hass.states.get(entity_id)
        if not state:
            continue
        # Try setting to min, max, and mid value if possible
        min_value = state.attributes.get("min")
        max_value = state.attributes.get("max")
        values = []
        if min_value is not None and max_value is not None:
            try:
                min_value = float(min_value)
                max_value = float(max_value)
                mid_value = (min_value + max_value) / 2
                values = [min_value, mid_value, max_value]
            except Exception:
                values = [state.state]
        else:
            values = [state.state]

        for value in values:
            await hass.services.async_call(
                "number",
                "set_value",
                {"entity_id": entity_id, "value": value},
                blocking=True,
            )
            await hass.async_block_till_done()
            updated_state = hass.states.get(entity_id)
            assert updated_state is not None
            snapshot.assert_match(updated_state)
