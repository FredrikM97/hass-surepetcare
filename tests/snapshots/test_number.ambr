# serializer version: 1
# name: test_number_set_and_snapshot
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl One',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
# name: test_number_set_and_snapshot.1
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl One',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
# name: test_number_set_and_snapshot.2
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl One',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
# name: test_number_set_and_snapshot.3
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl Two',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
# name: test_number_set_and_snapshot.4
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl Two',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
# name: test_number_set_and_snapshot.5
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl Two',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
# name: test_platform_setup_and_discovery[number.maui_matlada_old_target_weight_bowl_one-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Target weight Bowl One',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'target_weight',
    'unique_id': '269654-bowl_0_target_weight',
    'unit_of_measurement': 'g',
  })
# ---
# name: test_platform_setup_and_discovery[number.maui_matlada_old_target_weight_bowl_one-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl One',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
# name: test_platform_setup_and_discovery[number.maui_matlada_old_target_weight_bowl_two-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Target weight Bowl Two',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'target_weight',
    'unique_id': '269654-bowl_1_target_weight',
    'unit_of_measurement': 'g',
  })
# ---
# name: test_platform_setup_and_discovery[number.maui_matlada_old_target_weight_bowl_two-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl Two',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[number.maui_matlada_old_target_weight_bowl_one-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Target weight Bowl One',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'target_weight',
    'unique_id': '269654-bowl_0_target_weight',
    'unit_of_measurement': 'g',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[number.maui_matlada_old_target_weight_bowl_one-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl One',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_one',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[number.maui_matlada_old_target_weight_bowl_two-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Target weight Bowl Two',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'target_weight',
    'unique_id': '269654-bowl_1_target_weight',
    'unit_of_measurement': 'g',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[number.maui_matlada_old_target_weight_bowl_two-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Target weight Bowl Two',
      'max': 100.0,
      'min': 0.0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
      'unit_of_measurement': 'g',
    }),
    'context': <ANY>,
    'entity_id': 'number.maui_matlada_old_target_weight_bowl_two',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
